Core/SiteRating.swift

```diff
@@ -24,7 +24,9 @@ public class SiteRating {
     
     public var url: URL
     public var hasOnlySecureContent: Bool
-    public let domain: String
+    public var domain: String {
```

Comment:
The domain used to be guaranteed (e.g it was generated off a non-changing url and couldn't be optional as init would fail). Now that the url mutates, I would suggest deleting the domain altogether or as a minimum making it optional. Then the items which use it are forced to consider how to respond to no data rather than accidentally assuming that the non-optionlal means there is data.